### DISCUSSION TITLE
[已解决]当我保存为pcap 的时候error:opening Kprobe

### DISCUSSION THREAD

**Discussion by pythonmandev** [gojue/ecapture#543] [General]
./ecapture tls -b 2 -i eth0 可以正常跑起来，但是./ecapture tls -b 2 -m pcap -i eth0 -w ecapture.pcapng 就报错了
tls_2024/05/13 08:07:46 EBPFProbeOPENSSL        module run failed, [skip it]. error:couldn't start bootstrap manager error:1 error occurred:
        * error:opening Kprobe: creating tracefs event (arch-specific fallback for "tcp_sendmsg"): creating probe entry on tracefs: open /sys/kernel/tracing/kprobe_events: no such file or directory, funcName:tcp_sendmsg, isRet:false, section:kprobe/tcp_sendmsg, {UID:, EbpfFuncName:tcp_sendmsg}

看起来的原因是内核老了，5.15，更换为 android 14 后，内核为 6.1，可以正常工作

**Comment by cfc4n** [comment]
抱歉，我漏掉了这个问题。

看上去更像是5.15的android里，`tcp_sendmsg`  这个kprobe不是真正导出符号的名字，应该有个别名，你需要查看一下别名是什么，再手动更改代码中的符号名，重新编译。 之后就应该可以正常工作。 

----

Apologies, I missed this question. It seems more like in the Android 5.15, the `tcp_sendmsg` kprobe is not the actual exported symbol name; there should be an alias for it. You need to check what the alias is and manually change the symbol name in your code, then recompile. After that, it should work properly.
